src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -83,9 +87,28 @@ private static bool IsForEachProperty(IPropertySymbol symbol)
             IPropertySymbol symbol,
             Document document,
             SemanticModel semanticModel,
+            FindReferencesSearchOptions options,
             CancellationToken cancellationToken)
         {
-            var nameReferences = await FindReferencesInDocumentUsingSymbolNameAsync(symbol, document, semanticModel, cancellationToken).ConfigureAwait(false);
+            var nameReferences = await FindReferencesInDocumentUsingSymbolNameAsync(
+                symbol, document, semanticModel, cancellationToken).ConfigureAwait(false); 
+
+            if (options.AssociatePropertyReferencesWithSpecificAccessor)
+            {
+                // We want to associate property references to a specific accessor (if an accessor
+                // is being referenced).  Check if this reference would match an accessor. if so, do
+                // not add it.  It will be added by PropertyAccessorSymbolReferenceFinder.
+                var semanticFacts = document.GetLanguageService<ISemanticFactsService>();
+
+                nameReferences = nameReferences.WhereAsArray(loc =>
+                {
+                    var accessors = GetReferencedAccessorSymbols(
+                        semanticFacts, semanticModel, symbol, 
+                        loc.Location.FindNode(getInnermostNodeForTie: true, cancellationToken),
+                        cancellationToken);
+                    return accessors.Length == 0;
```

Comment:
>accessors.Length == 0 [](start = 27, length = 21)

The only case where `GetReferencedAccessorSymbols` returns zero accessors is in the `nameof` scenario. Could we just check that instead  (no need to check if written to or read from)?

Maybe the comment would be clearer if we said we'll only add property references that don't involve reading from or writing to the property (as those involve accessors, and thus are covered by the other finder).

Reply:
@jcouv What about XML crefs?